SAMtemplates/main_template.yaml

```diff
@@ -0,0 +1,337 @@
+    AllowedValues: [true, false]
+  EnableSplunk:
+    Type: String
+    Description: Whether to use splunk
+    Default: false
+    AllowedValues: [true, false]
+  LogLevel:
+    Type: String
+    Description: The log level to set in the lambda
+    Default: "INFO"
+  LogRetentionDays:
+    Type: Number
+    Description: How long to keep logs for
+    Default: 30
+    AllowedValues:
+      [
+        1,
+        3,
+        5,
+        7,
+        14,
+        30,
+        60,
+        90,
+        120,
+        150,
+        180,
+        365,
+        400,
+        545,
+        731,
+        1096,
+        1827,
+        2192,
+        2557,
+        2922,
+        3288,
+        3653,
+      ]
+
+Conditions:
+  ShouldUseMutualTLS: !Equals [true, !Ref EnableMutualTLS]
+  ShouldUseSplunk: !Equals [true, !Ref EnableSplunk]
+
+Resources:
+  # updatePrescriptionStatus lambda
+  UpdatePrescriptionStatusResources:
+    Type: AWS::Serverless::Application
+    Properties:
+      Location: lambda_resources.yaml
+      Parameters:
+        CloudWatchKMSKey: !ImportValue account-resources:CloudwatchLogsKmsKeyArn
+        LambdaName: !Sub "${AWS::StackName}-UpdatePrescriptionStatus"
+        LogRetentionDays: !Ref LogRetentionDays
+  UpdatePrescriptionStatus:
+    Type: AWS::Serverless::Function
+    Properties:
+      FunctionName: !Sub "${AWS::StackName}-UpdatePrescriptionStatus"
+      CodeUri: ../packages
+      Handler: 'updatePrescriptionStatus.handler'
+      Role: !GetAtt UpdatePrescriptionStatusResources.Outputs.LambdaRoleArn
+      Environment:
+        Variables:
+          TABLE_NAME: !Sub '${AWS::StackName}-PrescriptionStatusUpdate'
+          LOG_LEVEL: !Ref LogLevel
+      Events:
+        updatePrescriptionStatus:
+          Type: HttpApi
+          Properties:
+            ApiId: !Ref HttpApiGateway
+            Path: /
+            Method: post
+    Metadata: # Manage esbuild properties
+      BuildMethod: esbuild
+      BuildProperties:
+        Minify: true
+        Target: "es2020"
+        Sourcemap: true
+        tsconfig: updatePrescriptionStatus/tsconfig.json
+        EntryPoints:
+          - updatePrescriptionStatus/src/updatePrescriptionStatus.ts
+
+  # Prescription Status Update table
+  PrescriptionStatusUpdate:
+    Type: 'AWS::DynamoDB::Table'
+    Properties:
+      TableName: !Sub '${AWS::StackName}-PrescriptionStatusUpdate'
+      AttributeDefinitions:
+        - AttributeName: "RequestID"
+          AttributeType: "S"
+        - AttributeName: "PrescriptionID"
+          AttributeType: "S"
+        - AttributeName: "PatientNHSNumber"
+          AttributeType: "S"
+        - AttributeName: "PharmacyODSCode"
+          AttributeType: "S"
+      KeySchema:
+        - AttributeName: "RequestID"
+          KeyType: "HASH"
+        - AttributeName: "PrescriptionID"
+          KeyType: "RANGE"
+      ProvisionedThroughput: 
+        ReadCapacityUnits: "5"
+        WriteCapacityUnits: "5"
+      GlobalSecondaryIndexes:
```

Comment:
Do we need this many secondary indexes?

Reply:
We need Global Secondary Indexes for Prescription ID, Patient NHS Number and Pharmacy ODS Code.